server/tests/parts/ComparablePartsTests.js

```
@@ -0,0 +1,60 @@
+const { test_cpu, test_videocard, test_motherboard, test_memory } = require('../../utils/parts/test_parts.js')
+const { getComparableParts } = require('../../utils/parts/ComparableParts.js')
+
+const getComparableCPUs_test = async () => {
+    const comparableCPUs = await getComparableParts(test_cpu)
+    if (comparableCPUs.length > 0) {
+        console.log("getComparableParts_test-CPU: Passed")
+        return true
+    } else {
+        console.log("getComparableParts_test-CPU: Failed")
+        return false
+    }
+}
+
+const getComparableVideoCards_test = async () => {
+    const comparableVideoCards = await getComparableParts(test_videocard)
+    if (comparableVideoCards.length > 0) {
+        console.log("getComparableParts_test-VideoCard: Passed")
+        return true
+    } else {
+        console.log("getComparableParts_test-VideoCard: Failed")
+        return false
+    }
+}
+
+const getComparableMotherboards_test = async () => {
+    const comparableMotherboards = await getComparableParts(test_motherboard)
+    if (comparableMotherboards.length > 0) {
+        console.log("getComparableParts_test-Motherboard: Passed")
+        return true
+    } else {
+        console.log("getComparableParts_test-Motherboard: Failed")
+        return false
+    }
+}
+
+const getComparableMemorys_test = async () => {
+    const comparableMemorys = await getComparableParts(test_memory)
+    if (comparableMemorys.length > 0) {
+        console.log("getComparableParts_test-Memory: Passed")
+        return true
+    } else {
+        console.log("getComparableParts_test-Memory: Failed")
+        return false
+    }
+}
+
+const getComparableParts_test = async () => {
+    await getComparableCPUs_test()
+    await getComparableVideoCards_test()
+    await getComparableMotherboards_test()
+    await getComparableMemorys_test()
+}
+
+const running_tests = async () => {
+    await getComparableParts_test()
```

n00b q: how does this unit test run?
like are these supposed to be returning true?
if false, then means test failed?

---

@amotor95 
Printing "pass/fail" in the console isn't the standard approach for unit tests. One of the key goals of testing is to automatically detect regressions without relying on manual inspection. Ideally, a test runner should fail loudly if something breaks, and succeed silently when all is well.

Suggest using a proper test framework (e.g., Jest, Mocha, Vitest, etc.), and replacing console.log(...) with assertions like  
```
const result = await getComparableCPUs(xx, xx);
  expect(result).toEqual([
    // Expected mock CPU objects
  ]);

 // Or test each result falls within expected bounds
  for (const cpu of result) {
    expect(cpu.cores).toBeGreaterThanOrEqual(8 * 0.8);
    expect(cpu.cores).toBeLessThanOrEqual(8 * 1.2);
  }
```